sway/commands/resize.c

```diff
@@ -0,0 +1,297 @@
+#include <errno.h>
+#include <math.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#include <wlr/util/log.h>
+#include "sway/commands.h"
+#include "sway/tree/layout.h"
+#include "log.h"
+
+static const int min_sane_w = 100, min_sane_h = 60;
+
+enum resize_unit {
+	RESIZE_UNIT_PX,
+	RESIZE_UNIT_PPT,
+	RESIZE_UNIT_DEFAULT,
+	RESIZE_UNIT_INVALID,
+};
+
+enum resize_axis {
+	RESIZE_AXIS_HORIZONTAL,
+	RESIZE_AXIS_VERTICAL,
+	RESIZE_AXIS_INVALID,
+};
+
+static enum resize_unit parse_resize_unit(const char *unit) {
+	if (strcasecmp(unit, "px") == 0) {
+		return RESIZE_UNIT_PX;
+	}
+	if (strcasecmp(unit, "ppt") == 0) {
+		return RESIZE_UNIT_PPT;
+	}
+	if (strcasecmp(unit, "default") == 0) {
+		return RESIZE_UNIT_DEFAULT;
+	}
+	return RESIZE_UNIT_INVALID;
+}
+
+static enum resize_axis parse_resize_axis(const char *axis) {
+	if (strcasecmp(axis, "width") == 0 || strcasecmp(axis, "horizontal") == 0) {
+		return RESIZE_AXIS_HORIZONTAL;
+	}
+	if (strcasecmp(axis, "height") == 0 || strcasecmp(axis, "vertical") == 0) {
+		return RESIZE_AXIS_VERTICAL;
+	}
+	return RESIZE_AXIS_INVALID;
+}
+
+static int parallel_coord(struct sway_container *c, enum resize_axis a) {
+	return a == RESIZE_AXIS_HORIZONTAL ? c->x : c->y;
+}
+
+static int parallel_size(struct sway_container *c, enum resize_axis a) {
+	return a == RESIZE_AXIS_HORIZONTAL ? c->width : c->height;
+}
+
+static void resize_tiled(int amount, enum resize_axis axis) {
+	struct sway_container *parent = config->handler_context.current_container;
+	struct sway_container *focused = parent;
+	struct sway_container *sibling;
```

Comment:
Can be defined when used.

Reply:
It's used throughout the function, I think this is a reasonable spot for it.

Reply:
It's confusing because it makes the reader think the value is read outside the loop.

Reply:
fair enough